python/ray/rllib/optimizers/async.py

```diff
@@ -35,8 +35,7 @@ def step(self):
         # Note: can't use wait: https://github.com/ray-project/ray/issues/1128
         while gradient_queue:
             with self.wait_timer:
-                fut, e = gradient_queue[0]
-                gradient_queue = gradient_queue[1:]
+                fut, e = gradient_queue.pop(0)
```

Comment:
Isn't this backwards? You want to deque FIFO, not LIFO.

Reply:
I think it's right; if I understand you correctly - 
```
x = [1,2,3]
x.pop(0) == 1
```

Reply:
Oh ic, missed the zero arg